modules/infra/main.tf

```diff
@@ -28,12 +39,17 @@ resource "azurerm_resource_group" "pcf_resource_group" {
   location = "${var.location}"
 }
 
+resource "azurerm_resource_group" "pcf_network_rg" {
```

Comment:
You're correct, though since the RG create api works as create/update it doesn't matter. I could add a flag that, like `create_vnet` defaults to true.

Reply:
Ok that makes sense. We have a couple reasons that we think a `data` source might be better here.

1. I believe that a `data` source would error if the network resource group didn't already exist. This feels more like the expected behavior to me, since the intended use case for this is for using a pre-created network resource group.
2. How confident are you that the create/update won't overwrite anything in an existing resource group? For example, if there is a property `foo` that is set to `bar` in Azure, and `foo` defaults to the empty string in Terraform, I wouldn't want `foo` to be set to the empty string in Azure after running these templates.

Reply:
I was just thinking through scenarios and yeah I agree with you. There are two scenarios where this would cause problems: tags and location. I'll update this to use the same `data` depends on `resource` pattern.

Reply:
Thanks!